src/osd/OSD.cc

```diff
@@ -2403,13 +2403,29 @@ OSD::OSD(CephContext *cct_,
   trace_endpoint.copy_name(ss.str());
 #endif
 
+  // Determine scheduler type for this OSD
+  auto get_op_queue_type = [&conf = cct->_conf]() {
+    if (auto type = conf.get_val<std::string>("osd_op_queue");
+        type != "debug_random") {
+      return get_op_queue_type_by_name(type);
+    } else {
+      static const std::vector<std::string> index_lookup = { "mclock_scheduler",
+                                                             "wpq" };
```

Comment:
Use enums in the index -- there's no reason to use a string here only to convert it to an enum 3 lines later.

Reply:
Done